base/ryu/Ryu.jl

```diff
@@ -111,16 +120,16 @@ end
 
 function Base.show(io::IO, x::T, forceuntyped::Bool=false, fromprint::Bool=false) where {T <: Base.IEEEFloat}
     compact = get(io, :compact, false)::Bool
-    buf = Base.StringVector(neededdigits(T))
+    buf = Memory{UInt8}(undef, neededfloatdigits(T))
```

Comment:
Extra bytes are needed (for "Float16()") if "typed && T isa Float16"

Reply:
I included this in `neededfloatdigits(::Float16)`, but I guess it would make sense to separate it so we can allocate less when `!typed`